src/lib/middleware/rbac-middleware.ts

```diff
@@ -54,6 +54,9 @@ const rbacMiddleware = (
                 projectId = await featureToggleStore.getProjectId(featureName);
             } else if (permission === CREATE_FEATURE) {
                 projectId = projectId || req.body.project || 'default';
+            } else if (permission.endsWith('FEATURE_STRATEGY')) {
+                projectId = projectId || req.body.projectId;
+                environment = environment || req.body.environmentId;
```

Comment:
Two ideas here: 
- The permission accepts a callback that will extract the project and environment
- We standardize how we propagate these parameters

We can decide to close this PR in case we want to move to the second option. The first one would require some ideation. Meanwhile, the workaround could help define the proper permission [here](https://github.com/ivarconr/unleash-enterprise/blob/master/src/routes/segment-controller.ts#L63) so it's documented in OpenAPI

Reply:
I like this idea!

Reply:
I'm closing this now and moving to the backlog until I can spend sometime implementing that